### ISSUE TITLE
phase correction updates are very slow with shift key

### ISSUE THREAD

**Issue by targos** [cheminfo/nmrium#1154]
- with just click and drag, updates are fluid
- with shift + click and drag, updates are slow
- the only difference should be the steps of each update, but the same number of updates should occur every second, so the speed should not be affected

**Comment by hamed-musallam** [comment]
@targos 
I checked what is happing, what I noticed that the step with 1 has an effect on the spectrum bigger the 0.1 this way it seems to work in a smoother way.

then other things that we need to look for a solution, how we can prevent useModa,useAlert and useHelp from re-rendering the component when I drag to change ph0 and ph1 in manual phase correction.   

for example, I tried to memorize the Header component  but it's rendering every time i change the ph0 or ph1

**Comment by lpatiny** [comment]
> I checked what is happing, what I noticed that the step with 1 has an effect on the spectrum bigger the 0.1 this way it seems to work in a smoother way.

@hamed-musallam  Yes this is quite obvious but how can this impact the speed ? The calculation is expected to take the same time.

**Comment by targos** [comment]
> then other things that we need to look for a solution, how we can prevent useModa,useAlert and useHelp from re-rendering the component when I drag to change ph0 and ph1 in manual phase correction.
> for example, I tried to memorize the Header component but it's rendering every time i change the ph0 or ph1

I'll have a look at this tomorrow

**Comment by targos** [comment]
@hamed-musallam

I used a combination of the React devtools and [`use-what-changed`](https://github.com/cheminfo/nmrium/blob/master/README_DEV.md#debug-what-changed-to-trigger-a-re-render) to debug the rerenders of the Header component.

I fixed it for `useAlert` in https://github.com/cheminfo/nmrium/commit/67f4edfa8da8c86e46134590335c8b0d6d035977 and will let you do something similar with `useModal` and `useHelp`.

The problem is that the context was always changing because `<AlertProvider value={{ ... }}>` created a new object everytime (triggering a re-render of all components that call `useAlert`).
I fixed it by using a `useMemo` to only change the context if one of its values changes.
I also removed `alerts` from the context because it isn't used anywhere, and that allows to actually never re-render the alert context after the first render because now it only contains callback that never change.
